sim/simulation.py

```diff
@@ -0,0 +1,221 @@
+"""
+simulation.py
+
+Author: Matthew Yu, Array Lead (2020).
+Contact: matthewjkyu@gmail.com
+Created: 5/27/20
+Last Modified: 5/28/20
+Description: This Simulation class contains the state of the simulation. 
+    It  1) collects data from the source and mppt
+    and 2) displays it using matplotlib.
+"""
+import matplotlib
+import matplotlib.pyplot as plt
+from bisect import bisect
+import numpy as np
+import math
+
+class Simulation:
+    disp_cycle  = []
+    # source output
+    disp_vsrc   = []
+    disp_isrc   = []
+    disp_psrc   = []
+    # mppt output
+    disp_vref   = []
+    disp_pref   = []
+    # tunable parameters
+    disp_temp   = []
+    disp_irrad  = []
+    disp_load   = []
+    # stats
+    disp_pDiff  = [] # % diff from max power for cycle x
+    disp_pDiffA = [] # % of cycles > % diff from max power by cycle x
+
+    disp_open = False
+
+    fig = None
+    fig_num = 0
+    axs = None
+    twin_ax = None
+
+    def __init__(self, mppt_name):
+        """
+        init
+
+        Args:
+            - mppt_name (String): name of MPPT Algorithm to be displayed
+        """
+        plt.ion()
```

Comment:
I think it would be better to remove the interactive mode and then just add a .show() call in the display method. This way there isn't a graph in the way while parameters are still being selected.

Reply:
Noted. This code was originally to support stepping through each cycle to see change over time without reinstantiating the window, but that functionality has been removed as doing that for 300+ cycles was tedious at best.